### PR TITLE
Exclusive player UI skeleton

### LINKED ISSUE #119: Have an exclusive player UI

### PR DESCRIPTION
**Pull request by tsogtb**
closes #119 
Implements the exclusive player UI _skeleton_. (dependent on #123)

We have the buttons for:
- Broadcasting 
- Adding to a Beacon
- Adding to your Spotify playlists
- Ignoring (Blacklisting) 
- Voting

Would be nice to have buttons for:
- Comment section

Would be nice to display:
- The current score
- Was it picked up from a beacon? and if so, which?

### CODE REVIEW

**Comment by tsogtb** [comment]
image

First sketch looks like this, open to suggestions and button functions. Comments would be appreciated.

**Comment by IB-12** [comment]
I think it looks pretty good! It would be even better if the shuffle and repeat button weren't slightly off-axis compared to the buttons on the nav bar. And also should we consider using the same icons as Spotify to keep the UX as seamless as possible ? But other than that good job 👍

File: app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/SurroundWithMiniPlayer.kt [hunk 1]
```diff
@@ -58,12 +75,155 @@ fun SurroundWithMiniPlayer(displayPlayer: Boolean, screen: @Composable () -> Uni
...
+                            tint = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.size(50.dp))
+                      }
+                      IconButton(onClick = { /*TODO*/}) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.beacon_add_icon),
+                            contentDescription = "",
+                            tint = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.size(50.dp))
+                      }
```
Comment (saraanej) [inline app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/SurroundWithMiniPlayer.kt:100]: I would advise creating a function for displaying an iconButton to improve modularity of your code. It avoids repeating the same lines multiple times

File: app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/SurroundWithMiniPlayer.kt [hunk 2]
```diff
@@ -130,3 +290,44 @@ fun HandleProgressChanges(uiState: TrackListViewModel.UiState, progress: Mutable
...
+              modifier = Modifier.size(20.dp))
+        }
+        IconButton(onClick = { onVoteSelected(2) }, Modifier.size(20.dp)) {
+          Icon(
+              painter = painterResource(id = R.drawable.upvote_two_icon),
+              contentDescription = "",
+              tint =
+                  if (selectedVote.value == 2) Color.Red else MaterialTheme.colorScheme.onSurface,
+              modifier = Modifier.size(20.dp))
+        }
```
Comment (saraanej) [inline app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/SurroundWithMiniPlayer.kt:331]: I would give the same comment for these lines of code as well

File: app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/TrackListViewModel.kt
Comment (saraanej) [inline app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/TrackListViewModel.kt:1]: Don't hesitate to write comments explaining attributes if they're not straightforward to understand :)

**Review by saraanej — COMMENTED** [review]
Great deal of code, good job for the hard work ! Don't forget to write tests for your UI as you go for better test coverage. Had some little remarks but overall, everything looks good to me !

**Review by mbouaissi — APPROVED** [review]
Looks very good nice work! Maybe a small change regarding the code would be to divide ExclusivePlayer.kt into different files, but for this I let you choose, if you think it's better to have everything in one file it's fine for me. Otherwise great job, it looks very nice :+1:

**Pull request merged**